libraries/botbuilder-dialogs-adaptive/src/actions/httpRequest.ts

```diff
@@ -286,88 +287,109 @@ export class HttpRequest<O extends object = {}> extends Dialog<O> implements Htt
+                case ResponsesTypes.Activity:
+                    result.content = await response.json();
+                    dc.context.sendActivity(result.content as Activity);
+                    break;
+                case ResponsesTypes.Activities:
+                    result.content = await response.json();
+                    dc.context.sendActivities(result.content as Activity[]);
+                    break;
+                case ResponsesTypes.Json: {
+                    const content = await response.text();
+                    try {
+                        result.content = JSON.parse(content);
+                    } catch {
+                        result.content = content;
+                    }
+                    break;
+                }
+                case ResponsesTypes.Binary: {
+                    const buffer = await response.arrayBuffer();
+                    result.content = new Uint8Array(buffer);
+                    break;
                 }
-                break;
+                case ResponsesTypes.None:
+                default:
+                    break;
             }
-            case ResponsesTypes.Binary: {
-                const buffer = await response.arrayBuffer();
-                result.content = new Uint8Array(buffer);
-                break;
+
+            traceInfo.response = result;
+
+            // Write trace activity for http request and response values.
+            await dc.context.sendTraceActivity('HttpRequest', traceInfo, 'Microsoft.HttpRequest', this.id);
+
+            if (this.resultProperty) {
+                dc.state.setValue(this.resultProperty.getValue(dc.state), result);
             }
-            case ResponsesTypes.None:
-            default:
-                break;
-        }
 
-        traceInfo.response = result;
+            return await dc.endDialog(result);
+        } catch (err) {
+            if (err instanceof FetchError) {
+                const result = new Result();
+                result.statusCode = StatusCodes.NOT_FOUND;
+                result.content = err.message;
+                traceInfo.response = result;
 
-        // Write trace activity for http request and response values.
-        await dc.context.sendTraceActivity('HttpRequest', traceInfo, 'Microsoft.HttpRequest', this.id);
+                // Write trace activity for http request and response values.
+                await dc.context.sendTraceActivity('HttpRequest', traceInfo, 'Microsoft.HttpRequest', this.id);
 
-        if (this.resultProperty) {
-            dc.state.setValue(this.resultProperty.getValue(dc.state), result);
-        }
+                if (this.resultProperty) {
+                    dc.state.setValue(this.resultProperty.getValue(dc.state), result);
+                }
 
-        return await dc.endDialog(result);
+                return await dc.endDialog(result);
+            } else {
+                throw new Error();
```

Comment:
We should throw the same error we caught.
```suggestion
                throw err;
```